### PR TITLE
Make pysane Python3-compatible

### PR DESCRIPTION
**Pull request by manisandro**
These changes allow the pysane module to be compiled both as a python2 or python3 module.

### CODE REVIEW

**Comment by warsaw** [comment]
String exceptions are no longer supported in Python 3, so you need to do this too:
-   ErrorObject = PyString_FromString("_sane.error");
-   ErrorObject = PyErr_NewException("_sane.error", NULL, NULL);

File: Sane/_sane.c [hunk 1]
```diff
@@ -1261,17 +1271,40 @@
...
 
-	/* Create the module and add the functions */
-	m = Py_InitModule("_sane", PySane_methods);
+PyMODINIT_FUNC
+init_sane(void)
+{
+    /* Create the module and add the functions */
+    PyObject *m = Py_InitModule("_sane", PySane_methods);
+    if(!m)
+        return;
```
Comment (warsaw) [inline Sane/_sane.c:1302]: This has to return NULL

File: Sane/_sane.c [hunk 2]
```diff
@@ -28,6 +28,12 @@
 
 #include <sys/time.h>
 
+#if PY_MAJOR_VERSION >= 3
```
Comment (warsaw) [inline Sane/_sane.c:31]: Is the intention for Pillow to also provide Python 2 support?  If not, it's better to drop the #if and do the renames inline.

Comment (manisandro) [inline Sane/_sane.c:31]: I think that for the foreseeable future Pillow will also provide Python 2 support.

File: Sane/_sane.c (hunk 1 above)
Comment (manisandro) [inline Sane/_sane.c:1302]: That section of the code is for the python2 variant, where the module init function is void.

**Comment by fluggo** [comment]
This was cherry-picked. (Which I now realize was the wrong thing to do, but the damage is done.)

**Comment by manisandro** [comment]
Uhm, I actually had a patch ready with the suggested modifications (but can't commit it until this evening being behind an annoying corporate firewall atm). Should I post a new pull-request?

**Comment by manisandro** [comment]
Oh, you already made the changes. Ok, never mind - thanks!

**Pull request closed without merging**